packages/editor/CodeMirror/utils/searchExtension.ts

```diff
@@ -0,0 +1,40 @@
+import { EditorState, Extension } from '@codemirror/state';
+import { EditorView } from '@codemirror/view';
+import { EditorSettings, OnEventCallback } from '../../types';
+import getSearchState from './getSearchState';
+import { EditorEventType } from '../../events';
+import { search, searchPanelOpen, setSearchQuery } from '@codemirror/search';
+
+const searchExtension = (onEvent: OnEventCallback, settings: EditorSettings): Extension => {
+	const onSearchDialogUpdate = (state: EditorState) => {
+		const newSearchState = getSearchState(state);
+
+		onEvent({
+			kind: EditorEventType.UpdateSearchDialog,
+			searchState: newSearchState,
+		});
+	};
+
+	return [
+		search(settings.useExternalSearch ? {
+			createPanel(_editor: EditorView) {
+				return {
+					// The actual search dialog is implemented with react native,
+					// use a dummy element.
+					dom: document.createElement('div'),
+					mount() { },
+					destroy() { },
+				};
+			},
+		} : undefined),
+
+		EditorState.transactionExtender.of((tr) => {
+			if (tr.effects.some(e => e.is(setSearchQuery)) || searchPanelOpen(tr.state) !== searchPanelOpen(tr.startState)) {
+				onSearchDialogUpdate(tr.state);
+			}
```

Comment:
On mobile, this change leads to more information being sent over IPC while searching. This is because `onSearchDialogUpdate` fires an `UpdateSearchDialog` event, which now happens whenever the search query changes. Previously, `UpdateSearchDialog` events were fired only when the search dialog was opened/closed.